.github/workflows/main.yml

```diff
@@ -54,9 +54,9 @@ jobs:
             container_image: alpine:3.21
             host_dmd: ldmd2
           # macOS
-          - job_name: macOS 13 x64, DMD (latest)
-            os: macos-13
-            host_dmd: dmd
+          - job_name: macOS 14 arm64, LDC
+            os: macos-14
+            host_dmd: ldc
```

Comment:
Switching the host D compiler to LDC on these macOS arm64 runners means that a native arm64 LDC build is used, generating arm64 binaries. So the fresh DMD is built as a native arm64 executable - but still defaults to generating x86_64 code.

[Using a DMD host compiler yields a fresh x86_64 DMD build, which then passes all tests on the arm64 box. In that case, both the host compiler and the fresh DMD need to run in the Rosetta 2 emulator, as both are x86_64 executables.]